Adds demo instructions on windows (#68)
ADAH-EviDENce/evidence: README.md

[https://github.com/ADAH-EviDENce/evidence-gui](https://github.com/ADAH-EviDENce/evidence-gui)

> ## Running a demo on Windows

Reviewer: ```suggestion
## Running the demo on Windows system
```

Author: Changed it to "Running the demo on Windows"
I think adding 'system' would cause more confusion then clarification.

> Prerequisites:

Reviewer: Is WSL needed?

Author: no wsl needed

> - Docker desktop

Reviewer: Can you add a link?
https://docs.docker.com/docker-for-windows/install/

Author: good idea

> 

Reviewer: How would users test the Docker installation? Maybe test with hellow-world image?

Author: Added docker run hello-world instructions

> [Download](https://github.com/ADAH-EviDENce/WorkingTitleCloseReader/archive/master.zip) a copy of WorkingTitleCloseReader archive and extract its contents on your machine.

Reviewer: I like adding numbers to steps. It makes the separation more clear. Can you do that?

> Open a Powershell prompt:

Reviewer: Can we exaplain how we can start powershell?

Author: yeah why not, it could save our users a search
added it.

Change your current working directory to where you extracted the files. For instance:

```shell
cd C:\Users\JohnDoe\Downloads\WorkingTitleCloseReader-master\WorkingTitleCloseReader-master
```

The demo can be started with the following command:

```shell

> docker-compose --env-file getuigenverhalen.env up --build

Reviewer: Is docker-compose coming with docker-desktop?

Reviewer 2: Yep, see https://docs.docker.com/desktop/

Creating network "WorkingTitleCloseReader-master_default" with the default driver
Building server
Step 1/28 : FROM golang:1.12-stretch as buildserver
 ---> 563601c9e3b2
...
Creating WorkingTitleCloseReader-master_elasticsearch_1 ... done                                                                               Creating WorkingTitleCloseReader-master_indexer_1       ... error
ERROR: for WorkingTitleCloseReader-master_indexer_1  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for WorkingTitleCloseReader-master_server_1  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for indexer  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for server  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...
ERROR: Encountered errors while bringing up the project.
```

> The above error means that the docker containers cannot access the necessary folder. This can be solved by explicitly giving docker access to it. Do this by opening the Docker dashboard by right-clicking the docker icon in the Windows taskbar. Then, go to Settings/Resources/FILE SHARING and add the folder where you extracted the files before, and try running the command above again.

Reviewer: We need to reorganize this.
First explain what can cause a problem then show the error message. Finally, give the solution.

> Keep the Powershell window open during the demo.

Reviewer: Can you move this sentence to line 26?

For the demo we need to create a user. This can be done by opening a new powershell and run the following command in it:
```
invoke-webrequest -method Post -uri http://localhost:8080/users -body johndoe
```

> Note that the username, in our case 'johndoe', in the above command should be alpha-numerical and lowercase.

Reviewer: Can you mention that they can use another username. It may not be obvious for some users.

Author: changed sentence a bit to make this more clear.

> Browse to [http://localhost:8080/](http://localhost:8080/ui/search/).

Reviewer: Can you mention this is something you should do in your web browser? :)

[https://github.com/ADAH-EviDENce/evidence-gui](https://github.com/ADAH-EviDENce/evidence-gui)

## Running the demo on Windows

Prerequisites:

> - [Docker desktop](https://docs.docker.com/docker-for-windows/install/)

Reviewer 2: On the download page I must pick between stable or edge channel. Would be nice to document which channel to use.
```suggestion
- [Docker desktop](https://docs.docker.com/docker-for-windows/install/) from stable channel
```

### Step 1
First test that the docker installation is working. Open a Powershell prompt (press Windows+S and type Powershell) and run:
```shell
docker run hello-world
```
This should show a message that your Docker installation is working correctly. If so, we can proceed to the installation of WorkingTitleCloseReader, otherwise we suggest to check the steps in the [Docker installation guide](https://docs.docker.com/docker-for-windows/install/).

### Step 2
[Download](https://github.com/ADAH-EviDENce/WorkingTitleCloseReader/archive/master.zip) a copy of WorkingTitleCloseReader archive and extract its contents on your machine.

### Step 3
Open a Powershell prompt:
Change your current working directory to where you extracted the files. For instance:

```shell
cd C:\Users\JohnDoe\Downloads\WorkingTitleCloseReader-master\WorkingTitleCloseReader-master
```

The demo can be started with the below command. Keep this Powershell window open and running during the demo.

```shell

> docker-compose --env-file getuigenverhalen.env up --build

Reviewer 2: Use env var instead of file
```suggestion
$Env:EXPERIMENT="getuigenverhalen"
docker-compose up --build
```

Reviewer 2: ~~I installed Docker Desktop with wsl 2 integration enabled and the searches on the frontend are failing,. Seems to be same errors as in #77~~ Once #104 PR is merged the problem goes away.

ERROR: for WorkingTitleCloseReader-master_indexer_1  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for WorkingTitleCloseReader-master_server_1  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for indexer  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for server  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...
ERROR: Encountered errors while bringing up the project.
```

This can be solved by explicitly giving docker access to it. Do this by opening the Docker dashboard by right-clicking the docker icon in the Windows taskbar. Then, go to Settings/Resources/FILE SHARING and add the folder where you extracted the files before, and try running the command above again.

> For the demo we need to create a user, in our case 'johndoe'. This can be done by opening another powershell prompt and run the following command in it:
> ```
> invoke-webrequest -method Post -uri http://localhost:8080/users -body johndoe
> ```
> Note that any username you choose to use in the command above, must be alpha-numerical and lowercase.

Reviewer 2: When master branch has been merged into this branch this section is no longer needed.
```suggestion
For the demo we need to create a user, in our case 'johndoe'. This can be done by opening another powershell prompt and run the following command in it:
```
invoke-webrequest -method Post -uri http://localhost:8080/users -body johndoe
```
Note that any username you choose to use in the command above, must be alpha-numerical and lowercase.
```

[https://github.com/ADAH-EviDENce/evidence-gui](https://github.com/ADAH-EviDENce/evidence-gui)

## Running the demo on Windows

Prerequisites:

- [Docker desktop](https://docs.docker.com/docker-for-windows/install/)

### Step 1

First test that the docker installation is working. Open a Powershell prompt (press Windows+S and type Powershell) and run:

```shell
docker run hello-world
```

> This should show a message that your Docker installation is working correctly. If so, we can proceed to the installation of WorkingTitleCloseReader, otherwise we suggest to check the steps in the [Docker installation guide](https://docs.docker.com/docker-for-windows/install/).

Reviewer: ```suggestion
This should show a message that your Docker installation is working correctly. If so, we can proceed to the installation of WorkingTitleCloseReader, otherwise we suggest to check the steps in the [Docker troubleshooting page](https://docs.docker.com/docker-for-windows/troubleshoot/).
```

### Step 2

[Download](https://github.com/ADAH-EviDENce/evidence/archive/master.zip) a copy of evidence archive and extract its contents on your machine.

### Step 3

Open a Powershell prompt:
Change your current working directory to where you extracted the files. For instance:

```shell
cd C:\Users\JohnDoe\Downloads\evidence-master\evidence-master
```

The demo can be started with the below command. Keep this Powershell window open and running during the demo.

```shell
$Env:EXPERIMENT="demo"
docker-compose up --build
```

The command above downloads necessary Docker images, builds all the Docker images and starts the demo.

The command prints many log messages. If all goes well, the last lines of the output should be:

```shell
...
indexer_1        | Indexing done.
evidence-master_indexer_1 exited with code 0
```

> In some cases, the docker containers cannot access the necessary folder. In this case, you get output like the following:

Reviewer: Can you explain what are the necessary folders? 
Also what do you mean with ``some cases``?

Reviewer 3: My sggestion is to turn this argument around: "_if you get output like the following_"
```powershell
powershell stuff
```
_the docker containers cannot access the necessary folder C:\Windows\makes\everything\more\difficult._

Reviewer 2: Its windows 10 pro vs home, clearified in readme + clearified folder contents

```powershell
PS C:\Users\JohnDoe\evidence-master> docker-compose up --build
Creating network "evidence-master_default" with the default driver
Building server
Step 1/28 : FROM golang:1.12-stretch as buildserver
 ---> 563601c9e3b2
...
Creating evidence-master_elasticsearch_1 ... done                                                                               Creating evidence-master_indexer_1       ... error
ERROR: for evidence-master_indexer_1  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for evidence-master_server_1  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for indexer  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for server  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...
ERROR: Encountered errors while bringing up the project.
```

> This can be solved by explicitly giving docker access to it. Do this by opening the Docker dashboard by right-clicking the docker icon in the Windows taskbar. Then, go to Settings/Resources/FILE SHARING and add the folder where you extracted the files before, and try running the command above again.

Reviewer: ```suggestion
This can be solved by explicitly giving Docker access to the folder. You can do this by:
1. opening the Docker dashboard by right-clicking the Docker icon in the Windows taskbar 
2. go to Settings/Resources/FILE SHARING and add the folder where you extracted the files before
3. try running the command again:
```shell
$Env:EXPERIMENT="demo"
docker-compose up --build
```
```

### Step 4

> Go to the following URL in your webbrowser: [http://localhost:8080/](http://localhost:8080/ui/search/).

Reviewer: ```suggestion
Go to the following URL in your web browser: [http://localhost:8080/](http://localhost:8080/ui/search/).
```

### Step 5

> Once you done exploring the demo, you can stop the demo by selecting the powershell that is still running the demo and press Ctrl+C.

Reviewer: ```suggestion
Once you are done with exploring the demo, you can stop it by selecting the PowerShell that is still running the demo and press Ctrl+C.
```

[https://github.com/ADAH-EviDENce/evidence-gui](https://github.com/ADAH-EviDENce/evidence-gui)

## Running the demo on Windows

Prerequisites:

- [Docker desktop](https://docs.docker.com/docker-for-windows/install/)

### Step 1

First test that the docker installation is working. Open a Powershell prompt (press Windows+S and type Powershell) and run:

```shell
docker run hello-world
```

This should show a message that your Docker installation is working correctly. If so, we can proceed to the installation of evidence tool, otherwise we suggest to check the [Docker troubleshooting page](https://docs.docker.com/docker-for-windows/troubleshoot/).

### Step 2

[Download](https://github.com/ADAH-EviDENce/evidence/archive/master.zip) a copy of evidence archive and extract its contents on your machine.

### Step 3

Open a Powershell prompt:
Change your current working directory to where you extracted the files. For instance:

```shell

> cd C:\Users\JohnDoe\Downloads\evidence-master\evidence-master

Reviewer 3: ```suggestion
cd C:\Users\JohnDoe\Downloads\evidence
```
is probably easier. Note this will also affect the container names below.

Reviewer 2: This is the path you get when you use the Windows extractor with default settings. So I would like to keep it.

```

The demo can be started with the below command. Keep this Powershell window open and running during the demo.

```shell

> $Env:EXPERIMENT="demo"

Reviewer 3: ```suggestion
$Env:EXPERIMENT="getuigenverhalen"
```

Reviewer 2: Why? We have the demo experiment for the demo and this is the command to get the demo running.

Reviewer 2: wont fix

ERROR: for evidence-master_indexer_1  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for evidence-master_server_1  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for indexer  Cannot create container for service indexer: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...

ERROR: for server  Cannot create container for service server: status code not OK but 500: {"Message":"Unhandled exception: Filesharing has been cancelled",...
ERROR: Encountered errors while bringing up the project.
```

This can be solved by explicitly giving Docker access to the folder. You can do this by:

1. opening the Docker dashboard by right-clicking the Docker icon in the Windows taskbar
2. go to Settings/Resources/FILE SHARING and add the folder where you extracted the files before
3. try running the command again:

    ```shell

>     $Env:EXPERIMENT="demo"

Reviewer 3: ```suggestion
    $Env:EXPERIMENT="getuigenverhalen"
```

Reviewer 2: wontfix

    docker-compose up --build
    ```

### Step 4

Go to the following URL in your web browser: [http://localhost:8080/](http://localhost:8080/ui/search/).

### Step 5

Once you are done with exploring the demo, you can stop it by selecting the PowerShell that is still running the demo and press Ctrl+C.

> ## Generating a model from the corpus

Reviewer 3: I suggest moving all the instructions to get your docker/docker-compose working to a separate file, and link to it from the Prerequistites section. That way, the README can stay relatively clean.

Reviewer 2: we do not think its enough to put in a separate file.
